### PR TITLE
Raise defaults for preallocation in HTs optimisation

### PR DESCRIPTION
**Pull request by nickitat**
### Changelog category (leave one):
- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Default values for settings `max_size_to_preallocate_for_aggregation`, `max_size_to_preallocate_for_joins` were further increased to `10^12`, so the optimisation will be applied in more cases.

### CODE REVIEW

**Comment by robot-ch-test-poll** [comment]
*This is an automated comment for commit 7add1b73ef776341747c1069d717daed73b177f9 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72555/7add1b73ef776341747c1069d717daed73b177f9/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/a38700e5d1d0507873449e1c9d30e198f4c16920/docs_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: src/Core/Settings.cpp [hunk 1]
```diff
@@ -4312,14 +4312,14 @@ SELECT * FROM x_dist ORDER BY number ASC;
     DECLARE(Bool, collect_hash_table_stats_during_aggregation, true, R"(
 Enable collecting hash table statistics to optimize memory allocation
 )", 0) \
-    DECLARE(UInt64, max_size_to_preallocate_for_aggregation, 100'000'000, R"(
+    DECLARE(UInt64, max_size_to_preallocate_for_aggregation, 1'000'000'000'000, R"(
```
Comment (alexey-milovidov) [inline src/Core/Settings.cpp:4315]: Do we really target for a trillion, or use this as a "big enough" value?

Comment (nickitat) [reply src/Core/Settings.cpp:4315]: the latter. you think i should change it to `std::numeric_limits<UInt64>::max()`?

Comment (alexey-milovidov) [reply src/Core/Settings.cpp:4315]: This is ok, keep it :)

**Comment by nickitat** [comment]
`Stateless tests (debug) [1/2]` - #72566

**Pull request merged**